File: spec/python/test_nav_parent.py
import unittest

from nav_parent import NavParent

class TestNavParent(unittest.TestCase):
    def test_nav_parent(self):
        r = NavParent.from_file("src/nav.bin")

        self.assertEqual(r.header.qty_entries, 2)
        self.assertEqual(r.header.filename_len, 8)

        self.assertEqual(len(r.index.entries), 2)
        self.assertEqual(r.index.entries[0].filename, "FIRST___")
        self.assertEqual(r.index.entries[1].filename, "SECOND__")

File: spec/javascript/test_nav_parent.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('NavParent', 'src/nav.bin', function(r) {
    assert.equal(r.header.qtyEntries, 2);
    assert.equal(r.header.filenameLen, 8);

    assert.equal(r.index.entries.length, 2);
    assert.equal(r.index.entries[0].filename, "FIRST___");
    assert.equal(r.index.entries[1].filename, "SECOND__");
});
